packages/iov-bns/src/decode.ts

```diff
@@ -279,3 +286,42 @@ function parseRemoveAddressFromUsernameTx(
     },
   };
 }
+
+function ensureParticipants(
+  maybe: ReadonlyArray<codecImpl.multisig.IParticipant> | null | undefined,
+): ReadonlyArray<Participant> {
+  const participants = ensure(maybe, "participants");
+  return participants.map(participant => {
+    if (participant.signature === null || participant.power === null) {
+      throw new Error("invalid participant in participants");
+    }
+    return participant as Participant;
```

Comment:
Casts can be dangerous, especially when refactoring code later. Would this also work by making the map's return type explicit and let TypeScript check of `participant` matches the return type? Something like

```ts
  return participants.map((participant): Participant => {
    if (participant.signature === null || participant.power === null) {
      throw new Error("invalid participant in participants");
    }
    return participant;
  }
```

Reply:
Sadly not, even if I do the proper check and rule out `undefined`.